pkg/eval/eval.go

```diff
@@ -91,7 +91,11 @@ type Evaler struct {
 // NewEvaler creates a new Evaler.
 func NewEvaler() *Evaler {
 	builtin := builtinNs.Ns()
-	beforeChdirElvish, afterChdirElvish := vals.EmptyList, vals.EmptyList
+
+	newListVar := func(l vals.List) vars.PtrVar { return vars.FromPtr(&l) }
+	preExitHookElvish := newListVar(vals.EmptyList)
+	beforeChdirElvish := newListVar(vals.EmptyList)
+	afterChdirElvish := newListVar(vals.EmptyList)
```

Comment:
I don't like this change, it makes the `CallHook` calls below ugly: we need an unnecessary `.Get()`, and we lose type safety and have to use a type assertion.

Reply:
I thought the var needs to be passed around by reference while protected by a mutex. Is there a general rule for this, like which variables need to be put behind mutex?

Reply:
Oh that's a great point I missed, I'll keep your change.

Yes all _global_ variables need to be mutex-protected, and these are all global variables.